clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/DeployKubernetesAtomicOperation.groovy

```diff
@@ -59,10 +60,19 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
   ReplicationController deployDescription() {
     task.updateStatus BASE_PHASE, "Initializing creation of replication controller."
 
+
+    task.updateStatus BASE_PHASE, "Looking up provided namespace..."
+
     def credentials = description.kubernetesCredentials
+    def namespace = description.namespace ? description.namespace : "default"
+
+    if (!credentials.isRegisteredNamespace(namespace)) {
+      throw new KubernetesIllegalArgumentException("Namespace $namespace was not provided as a registered namespace within Clouddriver. Registered namespaces are ${credentials.getNamespaces()}.")
```

Comment:
What is the behavior if exactly one namespace is configured for the given account within clouddriver, it is not the `default` namespace, and the request elides the namespace attribute?

I also think the error message might be more helpful if s/within Clouddriver/for account $theAccountName.